client/state/signup/steps/site-type/actions.js

```diff
@@ -11,21 +10,9 @@ export function setSiteType( siteType ) {
 	};
 }
 
-export function submitSiteType( siteType, stepName = 'site-type' ) {
+export function submitSiteType( siteType, stepName ) {
 	return ( dispatch ) => {
 		dispatch( setSiteType( siteType ) );
-
-		let themeSlugWithRepo = undefined;
-		if ( 'site-type-with-theme' !== stepName ) {
```

Comment:
The step name would be `site-type-with-theme`, so the result is always falsy and we can remove it.